### PR TITLE
Add tab to show only errors in current line

### LINKED ISSUE #587: Only show Linter panel errors for selected line
I noticed that Linter recently updated to v1.0, and this is one absolutely essential feature I really miss. The pre-1.0 option to limit visible panel errors to the selected line totally rocks, as I prefer to minimize distraction within my editor. Could this be added to v1?

Thanks,
Caleb

### PR DESCRIPTION
**Pull request by josa42**
I added a third tab to the bottom which lets you filter the messages in the panel to show only error of the current line.

This is still work in progress. But I would like to know what you guys think of this idea.

screen-recording

Fixes #587.

### CODE REVIEW

**Comment by steelbrain** [comment]
@josa42 You might want to merge master in.

**Comment by k2b6s9j** [comment]
I'd be alright merging this in if the button was a configuration option disabled by default.

**Comment by k2b6s9j** [comment]
This could really be the solution for #587.

**Comment by steelbrain** [comment]
:+1: For an option that's disabled by default.

**Comment by steelbrain** [comment]
btw, You have accidently added `test.js` to the repo.

File: test.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+
+var tes
+
```
Comment (k2b6s9j) [inline test.js]: heh?

**Comment by josa42** [comment]
@steelbrain Ops I'll remove that. But like I said it's not done yet anyway.

**Comment by josa42** [comment]
I am done. I think it could be merged.

**Comment by steelbrain** [comment]
Looks good, gonna test it locally

**Comment by steelbrain** [comment]
Be advised that I pushed a typo fix to master, so You'll have to rebase.

**Comment by k2b6s9j** [comment]
:+1:

File: lib/linter-views.coffee [hunk 2]
```diff
@@ -110,9 +124,32 @@ class LinterViews
       )
       throw null
 
+  updateCurrentLine: (line) ->
+    if @_currentLine isnt line
+      @_currentLine = line
+      @_updateLineMessages()
+
+
+  _updateLineMessages: ->
```
Comment (steelbrain) [inline lib/linter-views.coffee]: I think we need LintMessages for only current editor, also changing active pane triggers `LinterViews::render`

Comment (josa42) [inline lib/linter-views.coffee]: I am not sure what you mean by than.

**Comment by AsaAyers** [comment]
3 tabs is way too many. As long as we're making one optional, lets just add options for each of them an you can have any combination of 0-3 tabs you'd like.

File: lib/linter-views.coffee (hunk 2 above)
Comment (steelbrain) [inline lib/linter-views.coffee]: You are currently collecting messages from eachLinterEditor, I had suggested that we collect only active's.

**Comment by steelbrain** [comment]
@AsaAyers It's off by default and is configurable by options.

**Comment by AsaAyers** [comment]
I understand that, Please add 2 more options and set them on by default to show the other two tabs. So that the people who want to only show errors from the current line don't have to have 3 tabs, 2 of which they'll never use. I'm not interested in the `Project` tab, I'm going to turn that one off

**Comment by josa42** [comment]
@AsaAyers I agree. Let's have an option for every tab.

File: lib/linter-views.coffee (hunk 2 above)
Comment (josa42) [inline lib/linter-views.coffee]: That makes sense I'll look into it.

**Comment by josa42** [comment]
I think it is ready to get merge now.

**Comment by steelbrain** [comment]
:clap: Gonna test it now

**Comment by steelbrain** [comment]
Looks great but can you make the first letter of the config values Uppercase? 'cause in settings `File` is better than `file`. Also any ideas why isn't the status icon position config not working anymore?

**Comment by josa42** [comment]
@steelbrain Sorry, I overlooked the statusIconPosition option while merging. I fixed it.

**Comment by k2b6s9j** [comment]
:shipit:

**Comment by robneu** [comment]
Just wanted to say :heart: :heart_eyes: :100:

**Comment by bjmiller** [comment]
This is terrific.  Excellent work, @josa42 and everyone!

**Comment by steelbrain** [comment]
:clap: :rose:

**Comment by code-tree** [comment]
Thanks heaps for this feature. Though, is there a way to set which mode is activated when Atom is opened? I'd like to have all three tabs visible, but for Atom to default to the line mode when it opens (it currently seems to default to the file mode?)

**Comment by steelbrain** [comment]
@code-tree IIRC the last tab is persisted per project as part of state

**Comment by code-tree** [comment]
@steelbrain Ok thanks, I just realised it is probably losing its state because of a hack in my config for a different bug fix. Sorry for the confusion.

**Pull request merged**